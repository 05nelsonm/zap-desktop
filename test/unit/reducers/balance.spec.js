import balanceReducer, { GET_BALANCE, RECEIVE_BALANCE } from 'reducers/balance'

describe('reducers', () => {
  describe('balanceReducer', () => {
    it('should handle initial state', () => {
      expect(balanceReducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle GET_BALANCE', () => {
      expect(balanceReducer(undefined, { type: GET_BALANCE })).toMatchSnapshot()
    })

    it('should handle RECEIVE_BALANCE', () => {
      const walletBalance = {
        total_balance: 1,
        confirmed_balance: 1,
        unconfirmed_balance: 1
      }
      const channelBalance = 1
      expect(
        balanceReducer(undefined, { type: RECEIVE_BALANCE, walletBalance, channelBalance })
      ).toMatchSnapshot()
    })

    it('should handle unknown action type', () => {
      expect(balanceReducer(undefined, { type: 'unknown' })).toMatchSnapshot()
    })
  })
})
