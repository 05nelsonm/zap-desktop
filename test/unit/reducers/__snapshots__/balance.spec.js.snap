// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers balanceReducer should handle GET_BALANCE 1`] = `
Object {
  "balanceLoading": true,
  "channelBalance": null,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;

exports[`reducers balanceReducer should handle RECEIVE_BALANCE 1`] = `
Object {
  "balanceLoading": false,
  "channelBalance": 2,
  "walletBalance": 1,
  "walletBalanceConfirmed": 1,
  "walletBalanceUnconfirmed": 1,
}
`;

exports[`reducers balanceReducer should handle initial state 1`] = `
Object {
  "balanceLoading": false,
  "channelBalance": null,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;

exports[`reducers balanceReducer should handle unknown action type 1`] = `
Object {
  "balanceLoading": false,
  "channelBalance": null,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;
